Issue 9306 related

bridiver opened the pull request:
Submitter Checklist:
@cezaraugusto I did some related changes for #9306 that turned out not to be necessary, but are still good to have and can go into 0.19. Still need to check tests, but wanted to get this out there to look at

- [ ] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:


Reviewer Checklist:

Tests


- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

bsclifton commented on app/browser/reducers/tabsReducer.js:
```diff
@@ -67,6 +67,11 @@ const updateActiveTab = (state, closeTabId) => {
+    nextTabId = tabState.get('tabs').first()
+  }
+
   if (nextTabId !== tabState.TAB_ID_NONE) {
```
When I wrote the unit tests for this, I was unable to find a way to hit this condition.  When you originally pulled the logic out of frameStateUtil and moved here, you had a similar block.  Since it wasn't possible to hit, I had removed it.  Did you see a case where this runs?

bridiver commented:
This PR was changes I made that turned out not to be necessary for the fix and this is just leftover. It's actually not correct anyway because it should get the first tab in the window and that is already covered by "next" so this isn't needed

> On Jun 17, 2017, at 11:13 PM, Brian Clifton <notifications@github.com> wrote:
> 
> @bsclifton commented on this pull request.
> 
> In app/browser/reducers/tabsReducer.js:
> 
> > @@ -67,6 +67,11 @@ const updateActiveTab = (state, closeTabId) => {
>      }
>    }
>  
> +  // fall back to first tab
> +  if (nextTabId === tabState.TAB_ID_NONE) {
> +    nextTabId = tabState.get('tabs').first()
> +  }
> +
>    if (nextTabId !== tabState.TAB_ID_NONE) {
> When I wrote the unit tests for this, I was unable to find a way to hit this condition. When you originally pulled the logic out of frameStateUtil and moved here, you had a similar block.  Since it wasn't possible to hit, I had removed it. Did you see a case where this runs?
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cezaraugusto commented:
@bridiver are we following the dumb/smart components convention where util files are made to host the logic, leaving the main component just for presentational purposes? 

I'm changing the way we preview tabs in #8852 and if this turns to be accepted first I'll need to re-work the logic

bridiver commented:
the logic should be in the reducers (which can use state/util files) and yes, the components should be as "dumb" as possible. Ideally action should be things like `tabClicked` or `tabMouseEnter` letting the reducer to decide what actually needs to be done in response

bridiver commented:
@cezaraugusto it looks like this overrides some recent changes you made to the component. Can you make sure this covers whatever changes you made?

cezaraugusto commented:
@bridiver yes, thanks for the reminder. I'll make sure to update your changes. 

#8852 will also change it so keeping an eye anyway on which goes first.

bridiver commented:
I think #8852 is actually moving in the wrong direction with `setPreviewMode`. "setter" actions should be deprecated and we don't want to call multiple actions together. `setPreviewMode` doesn't provide any new information that `setPreviewFrame` doesn't already provide. I think better to merge this first (assuming there are no issues with it) and then make other changes

bridiver commented:
there is an issue with this PR that I am fixing right now

bridiver commented:
updated @cezaraugusto 
I really think we should merge this one first because it's only going to get more complicated to merge the functionality if we keep adding things to the component

cezaraugusto commented:
@bridiver btw once we're good to go I'm going to accept this one before the land of #8852 as per your request. That PR will be included in 0.18.x per last weekly triage meeting, so I'm going to move this to the same milestone as well.

cezaraugusto commented on app/renderer/components/tabs/tabPage.js:
```diff
@@ -32,23 +32,6 @@ class TabPage extends React.Component {
     this.onMouseLeave = this.onMouseLeave.bind(this)
```
I think those should be removed too

cezaraugusto replied:
maybe remove it all given other handlers are not bound to the constructor but curious if that affects reduxComponent

bridiver replied:
actually I'm surprised this doesn't cause an error, but I'll remove it

cezaraugusto replied:
btw there's no listener for tab page previews anymore and I can't preview a tab set.

bridiver replied:
ah, I think I accidentally grouped them together. Is there a test for it that is failing or did you just find it by manual testing?

cezaraugusto replied:
yes, you can try `npm run test -- --grep="tab pages tab page previews hovering over a tab page changes it"`

luixxiul commented:
@cezaraugusto I added the QA steps for the tab preview issue to the 1st post based on your feedback here: https://github.com/brave/browser-laptop/pull/9542#pullrequestreview-45567818. Please correct them if they are not right (I replaced `fourth tab` with `3rd tab` as there should be only 3 tabs opened).

cezaraugusto commented:
@luixxiul awesome, thanks for that
